clients/dataset/dataset.go

```diff
@@ -83,60 +43,42 @@ var _ error = ErrInvalidDatasetAPIResponse{}
 
 // Client is a dataset api client which can be used to make requests to the server
 type Client struct {
-	cli           RHTTPClient
-	ctxcli        RCHTTPClient
-	url           string
-	internalToken string
+	cli common.RCHTTPClienter
+	url string
 }
 
-// New creates a new instance of Client with a given dataset api url
-func New(datasetAPIURL string) *Client {
+// NewAPIClient creates a new instance of Client with a given dataset api url and the relevant tokens
+func NewAPIClient(datasetAPIURL, serviceToken, xDownloadServiceToken, florenceToken string) *Client {
 	return &Client{
-		cli:    rhttp.DefaultClient,
-		ctxcli: rchttp.DefaultClient,
-		url:    datasetAPIURL,
-	}
-}
-
-// SetInternalToken will set an internal token to use for the dataset api
-func (c *Client) SetInternalToken(token string) {
-	c.internalToken = token
-}
-
-func (c *Client) setRequestHeaders(req *http.Request, cfg ...Config) {
-	if len(cfg) > 0 {
-		req.Header.Set(authTokenHeader, cfg[0].InternalToken)
-		req.Header.Set(authorizationHeader, cfg[0].AuthToken)
-		req.Header.Set(xDownloadServiceHeader, cfg[0].XDownloadServiceToken)
-		req.Header.Set(florenceToken, cfg[0].FlorenceToken)
-	} else if len(c.internalToken) > 0 {
-		req.Header.Set("Internal-token", c.internalToken)
+		cli: rchttp.ClientWithDownloadServiceToken(
+			rchttp.ClientWithFlorenceToken(
+				rchttp.ClientWithServiceToken(nil, serviceToken),
+				florenceToken,
+			),
+			xDownloadServiceToken,
+		),
+		url: datasetAPIURL,
 	}
 }
 
 // Healthcheck calls the healthcheck endpoint on the api and alerts the caller of any errors
 func (c *Client) Healthcheck() (string, error) {
-	resp, err := c.cli.Get(c.url + "/healthcheck")
+	ctx := context.Background()
```

Comment:
Do we want to allow context to be passed into the healthcheck for cancellation? (possibly a future enhancement rather than something to add right now)

Reply:
I am leaning towards: _future!_